R/nSequences.R

```diff
@@ -0,0 +1,142 @@
+#'    listData = list(
+#'        msLevel = c(1L, 2L, 2L, 2L, 2L, 2L, 2L, 2L, 2L, 2L, 2L, 2L,
+#'                    2L, 2L, 2L, 2L, 2L, 2L, 2L, 1L, 2L, 2L, 2L, 2L,
+#'                    2L, 2L, 2L, 2L, 2L, 2L, 2L, 2L, 2L, 2L, 2L, 2L,
+#'                    2L, 1L, 2L, 2L, 2L, 2L, 2L, 2L, 2L, 2L, 2L, 2L,
+#'                    2L, 2L, 2L, 1L, 2L, 2L, 2L, 2L, 2L, 2L, 2L, 2L,
+#'                    2L, 2L, 2L, 2L, 2L, 2L, 2L, 2L, 2L, 2L, 1L, 2L,
+#'                    2L, 2L, 2L, 2L, 2L, 2L, 2L, 2L, 2L, 2L, 2L, 2L,
+#'                    2L, 2L),
+#'        acquisitionNum = 8975:9060,
+#'        precScanNum = c(NA, 8956L, 8956L, 8956L, 8956L, 8956L, 8956L,
+#'                        8956L, 8956L, 8956L, 8956L, 8956L, 8956L,
+#'                        8956L, 8956L, 8956L, 8956L, 8956L, 8956L, NA,
+#'                        8975L, 8975L, 8975L, 8975L, 8975L, 8975L,
+#'                        8975L, 8975L, 8975L, 8975L, 8975L, 8975L,
+#'                        8975L, 8975L, 8975L, 8975L, 8975L, NA, 8994L,
+#'                        8994L, 8994L, 8994L, 8994L, 8994L, 8994L,
+#'                        8994L, 8994L, 8994L, 8994L, 8994L, 8994L, NA,
+#'                        9012L, 9012L, 9012L, 9012L, 9012L, 9012L,
+#'                        9012L, 9012L, 9012L, 9012L, 9012L, 9012L,
+#'                        9012L, 9012L, 9012L, 9012L, 9012L, 9012L, NA,
+#'                        9026L, 9026L, 9026L, 9026L, 9026L, 9026L,
+#'                        9026L, 9026L, 9026L, 9026L, 9026L, 9026L,
+#'                        9026L, 9026L, 9026L),
+#'        sequence = c(NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA,
+#'                     NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA,
+#'                     "LSEHATAPTR", NA, NA, NA, NA, NA, NA, NA,
+#'                     "EGSDATGDGTK", NA, NA, "NEDEDSPNK", NA, NA, NA,
+#'                     NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA,
+#'                     NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA,
+#'                     NA, NA, NA, NA, NA, NA, NA, NA, NA, "GLTLAQGGVK",
+#'                     NA, NA, NA, NA, "STLPDADRER", NA, NA, NA, NA, NA,
+#'                     NA, NA, NA)),
+#'    elementType = "ANY", elementMetadata = NULL, metadata = list())
+#'
+#' sp <- Spectra(spdf)
+#'
+#' ## We have in this data 5 MS1 and 81 MS2 scans
+#' table(msLevel(sp))
+#'
+#' ## The acquisition number of the MS1 scans
+#' acquisitionNum(filterMsLevel(sp, 1))
+#'
+#' ## And the number of MS2 scans with precursor ions selected
+#' ## from MS1 scans (those in the data and others)
+#' table(precScanNum(sp))
+#'
+#' ## Count number of sequences/identifications per scan
+#' sp <- nSequences(sp)
+#'
+#' ## MS2 scans either lead to an identification (5 instances) or none
+#' ## (76). Among the five MS1 scans in the experiment, 3 lead to MS2
+#' ## scans being matched to no peptides and two MS1 scans produced two
+#' ## and three PSMs respectively.
+#' table(sp$nSequences, sp$msLevel)
+nSequences <- function(object,
```

Comment:
Using a variable/name `sequence` for an annotation seems to me not straight forward - in small molecule MS we don't have sequences but chemical formulas or similar. Thus, what about *generalizing* this and using the term *annotation* instead of *sequence* and calling this function `countAnnotations`?